lib/rubocop/cop/performance/hash_each_methods.rb

```diff
@@ -55,17 +67,32 @@ def autocorrect(node)
           caller, first_method = *receiver
 
           lambda do |corrector|
-            if receiver.method?(:hash)
-              correct_hash(node, corrector)
-            else
+            if first_method.eql?(:keys) || first_method.eql?(:values)
+              return correct_implicit(node, corrector) if receiver
```

Comment:
When its a plain_each match, then we check if receiver is nil. 
When its a key_value match, I have to check if the main receiver of (key || value) is nil. 

I have changed it to `receiver.receiver.nil?`. Not sure if there is a better node extension method that can be used in place here?